source/Planet.h

```diff
@@ -154,6 +160,10 @@ class Planet {
 
 	std::set<const Sale<Ship> *> shipSales;
 	std::set<const Sale<Outfit> *> outfitSales;
+
+	mutable CustomSale customSale;
+	mutable CustomSale visibleCustomSale;
+	mutable int cachedDays = -1;
```

Comment:
Storing this stuff in Planet doesn't seem desirable.

Could determining item availability and pricing not be done in the relevant ShopPanel constructor based on the global lists instead of trying to maintain copied lists in the Planets?

Reply:
That makes a lot of sense actually.

Reply:
Wait no this is used by the map outfitter and shipyard too
Issue is customsales with differemt selltypes dont add up although mb I could fix that by making the biggest priority taken into account